src/DeploymentProvider/AzureSpringCloudDeploymentProvider.ts

```diff
@@ -0,0 +1,207 @@
+                break;
+            }
+
+            case Actions.setProduction: {
+                await this.performSetProductionAction();
+                break;
+            }
+
+            case Actions.deleteStagingDeployment: {
+                await this.performDeleteStagingDeploymentAction();
+                break;
+            }
+
+            default:
+                throw Error('UnknownOrUnsupportedAction' + this.params.Action);
+        }
+    }
+
+    private async performDeleteStagingDeploymentAction() {
+        console.log('Delete staging deployment action');
+        const deploymentName = await dh.getStagingDeploymentName(this.client, this.params);
+        this.params.DeploymentName = deploymentName;
+        console.log(`Action for service ${this.params.ServiceName} app ${this.params.AppName} deployment ${deploymentName}`);
+        if (deploymentName) {
+            await dh.deleteDeployment(this.client, this.params);
+        } else {
+            throw Error('NoStagingDeploymentFound');
+        }
+        console.log('Delete staging deployment action successful');
+        return deploymentName;
+    }
+
+    private async performSetProductionAction() {
+        console.log('Set production action');
+        let deploymentName: string;
+        if (this.params.UseStagingDeployment) {
+            core.debug('Targeting inactive deployment');
+            deploymentName = await dh.getStagingDeploymentName(this.client, this.params);
+            this.params.DeploymentName = deploymentName;
+            if (!deploymentName) { //If no inactive deployment exists, we cannot continue as instructed.
+                throw Error('NoStagingDeploymentFound');
+            }
+        }
+        else {
+            //Verify that the named deployment actually exists.
+            deploymentName = this.params.DeploymentName;
+            let existingStagingDeploymentName: string = await dh.getStagingDeploymentName(this.client, this.params);
+            if (deploymentName != existingStagingDeploymentName) {
+                throw Error('StagingDeploymentWithNameNotExist:' + deploymentName);
+            }
+        }
+        console.log(`Action for service ${this.params.ServiceName} app ${this.params.AppName} deployment ${deploymentName}`);
+        await dh.setActiveDeployment(this.client, this.params);
+        console.log('Set production action successful');
+    }
+
+    private async performDeployAction() {
+        console.log('Deploy action');
+
+        let sourceType: string = this.determineSourceType(this.params.Package);
+
+        //If uploading a source folder, compress to tar.gz file.
+        let fileToUpload: string = sourceType == SourceType.SOURCE_DIRECTORY ?
+            await this.compressSourceDirectory(this.params.Package.getPath()) :
+            this.params.Package.getPath();
+
+
+        let deploymentName: string;
+        if (this.params.UseStagingDeployment) {
```

Comment:
Add some logs to the critical branch decisions. And also the followed `deploymentName` being resolved.